Prepare release 2.2.0

ggabernet opened the pull request:
<!--
# nf-core/epitopeprediction pull request

Many thanks for contributing to nf-core/epitopeprediction!

Please fill in the appropriate checklist below (delete whatever is not relevant).
These are the most common things requested on pull requests (PRs).

Remember that PRs should be made against the dev branch, unless you're preparing a pipeline release.

Learn more about contributing: [CONTRIBUTING.md](https://github.com/nf-core/epitopeprediction/tree/master/.github/CONTRIBUTING.md)
-->

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
  - [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/epitopeprediction/tree/master/.github/CONTRIBUTING.md)
  - [ ] If necessary, also make a PR on the nf-core/epitopeprediction _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

jfy133 commented on modules/local/csvtk_concat.nf:
```diff
@@ -1,7 +1,7 @@
 process CSVTK_CONCAT {
```
Any reason why this and split are local modules not the official nf-core ones? If you just want the sort command I would just patch the module with `nf-core modules patch` rather than having an entirely separate 'cusotm' local one

jfy133 replied:
Sounds good!

jfy133 reviewed (commented):
General comment as I started to say this on every `bin/` script: please add author and license to all of them :grimacing: 

Also that `epaa.py` script is rather chunky... I would recommend pushing it to a standalone package on bioconda or something - I ended up not reviewing it because it was full of black formatting - so I'm trusting you the VEP addition is there ;)

Otherwise LGTM

jfy133 commented:
@ggabernet shouldn't you add yourself to the authorlist if you're opening it

FriederikeHanssen reviewed (commented):
Some additional comments:

- There is a wild mixture of versions being reported in csv, txt and yml files. I haven't tested if this has any impact on the version collecting
- the versions for the external tools are not collected as far as I can see (EPYTOPE_CHECK_REQUESTED_MODELS and below, also for EPYTOPE_PEPTIDE_PREDICTION_PEP and below )
- groupTuple is not used with `size` parameter or groupKey, i.e line 406, causing stalling of the workflow

ggabernet commented on bin/check_samplesheet.py:
```diff
@@ -78,13 +77,15 @@ def validate_and_transform(self, row):
     def _validate_first(self, row):
         """Assert that the first FASTQ entry is non-empty and has the right format."""
-        assert len(row[self._first_col]) > 0, "At least the first FASTQ file is required."
+        if len(row[self._first_col]) <= 0:
```
yes those are unused functions in the `check_samplesheet.py` but they are part of the template, so I don't see an issue leaving them here

ggabernet commented on modules/local/cat_files.nf:
```diff
@@ -1,7 +1,7 @@
-    conda (params.enable_conda ? "conda-forge:sed=4.8" : null)
+    conda "conda-forge:sed=4.8"
     container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
         'https://depot.galaxyproject.org/singularity/cat:5.2.3--hdfd78af_1' :
         'quay.io/biocontainers/cat:5.2.3--hdfd78af_1' }"
```
I've just added the software version collection for cat, and opened an issue on using the nf-core module for this in a future release

ggabernet commented:
@jfy133 and @FriederikeHanssen , thanks for your reviews! I've addressed most comments in a separate PR:
https://github.com/nf-core/epitopeprediction/pull/194

jfy133 commented on CHANGELOG.md:
```diff
@@ -3,6 +3,27 @@
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## v2.2.0 - WaldhaeuserOst - 2023-03-01
```
Well that brings back memories of my Master's dorms...

ggabernet replied:
```suggestion
## v2.2.0 - WaldhaeuserOst - 2023-03-03
```

jfy133 reviewed (approved):
The license headers text should have the years I believe, but otherwise they look good :)

FriederikeHanssen reviewed (approved):
I agree with James. Otherwise lgtm 🚀

ggabernet commented on bin/gen_peptides.py:
```diff
@@ -1,4 +1,5 @@
 #!/usr/bin/env python
+# Written by Sabrina Krakau, Christopher Mohr and released under the MIT license.
```
```suggestion
# Written by Sabrina Krakau, Christopher Mohr and released under the MIT license (2022).
```

ggabernet commented on bin/merge_jsons.py:
```diff
@@ -1,4 +1,6 @@
 #!/usr/bin/env python
+# Written by Sabrina Krakau, Christopher Mohr, Gisela Gabernet and released under the MIT license.
```
```suggestion
# Written by Sabrina Krakau, Christopher Mohr, Gisela Gabernet and released under the MIT license (2022).
```

ggabernet commented on bin/split_peptides.py:
```diff
@@ -1,26 +1,32 @@
 #!/usr/bin/env python
+# Written by Sabrina Krakau, Christopher Mohr and released under the MIT license.
```
```suggestion
# Written by Sabrina Krakau, Christopher Mohr and released under the MIT license (2022).
```

ggabernet commented on bin/split_vcf_by_variants.py:
```diff
@@ -1,4 +1,5 @@
 #!/usr/bin/env python
+# Written by Christopher Mohr and released under the MIT license.
```
```suggestion
# Written by Christopher Mohr and released under the MIT license (2022).
```

ggabernet commented on bin/epaa.py:
```diff
@@ -1,4 +1,5 @@
 #!/usr/bin/env python
+# Written by Christopher Mohr and released under the MIT license.
```
```suggestion
# Written by Christopher Mohr and released under the MIT license (2022).
```
